index.d.ts

```diff
@@ -23,9 +23,20 @@ import { Readable } from 'stream'
 type Mercurius = typeof mercurius
 
 declare namespace mercurius {
-  export interface PubSub {
-    subscribe<TResult = any>(topics: string | string[]): Promise<Readable & AsyncIterableIterator<TResult>>;
-    publish<TResult = any>(event: { topic: string; payload: TResult }, callback?: () => void): void;
+  export type PubSubPublishArgsByKey = {
+    [key: string]: any;
+  };
+
+  export interface PubSub<
+    TPubSubPublishArgsByKey extends PubSubPublishArgsByKey,
+  > {
```

Comment:
Unless you do this, this is a breaking change, as `PubSub` will require `TPubSubPublishArgsByKey` to be specified:

```suggestion
  export interface PubSub<
    TPubSubPublishArgsByKey extends PubSubPublishArgsByKey = PubSubPublishArgsByKey,
  > {
```

Reply:
Thanks, fixed it.